block/aggregation.go

```diff
@@ -49,48 +49,92 @@ func (m *Manager) lazyAggregationLoop(ctx context.Context, blockTimer *time.Time
 	lazyTimer := time.NewTimer(0)
 	defer lazyTimer.Stop()
 
+	// Track if we've received transactions but haven't built a block yet
+	pendingTxs := false
+
 	for {
 		select {
 		case <-ctx.Done():
 			return
 
 		case <-lazyTimer.C:
+			m.buildingBlock = false
+			pendingTxs = false
+			m.produceBlockLazy(ctx, "lazy_timer", lazyTimer, blockTimer)
+
 		case <-blockTimer.C:
-		}
+			if pendingTxs {
+				m.buildingBlock = true
+				m.produceBlockLazy(ctx, "tx_collection_complete", lazyTimer, blockTimer)
+				pendingTxs = false
```

Comment:
not sure if we should call it `pendingTxs`, since txs could still be pending even after we finish building a block since a block only consumes one batch